tools/shared.py

```diff
@@ -2678,6 +2678,8 @@ def path_to_system_js_libraries_for_settings(link_settings):
       system_js_libraries += ['library_egl.js', 'library_webgl.js']
     if 'USE_WEBGL2=1' in link_settings:
       system_js_libraries += ['library_webgl2.js']
+    if 'USE_WEBGL2_COMPUTE=1' in link_settings:
+      system_js_libraries += ['library_webgl2.js', 'library_webgl2_compute.js']
```

Comment:
webgl2.js may be included twice if the user specifies both WEBGL2 and WEBGL2_COMPUTE, is that a problem?

Reply:
At least in current implementation it's not a problem. `USE_WEBGL2` and `USE_WEBGL2_COMPUTE` are mutual exclusive and checked elsewhere. But to be safe, I'm happy to renew the PR again and change `system_js_libraries` to an instance of `set()`.

(It seems that `library_webgl2.js` is already included twice: https://github.com/emscripten-core/emscripten/blob/4ce1300f41c9d29de08c7f65a34faaff275322d6/tools/shared.py#L2682-L2683 and https://github.com/emscripten-core/emscripten/blob/4ce1300f41c9d29de08c7f65a34faaff275322d6/src/modules.js#L131-L133)

Reply:
It's probably fine to leave it as is. There is some wasted compiler time from these duplications, but it doesn't lead to errors. We should remove the duplicates, though - I'll remove that one in tools/shared.py.

Reply:
Are there still duplicates to remove in this PR?